loopd/swapclient_server.go

```diff
@@ -1154,6 +1160,12 @@ func validateLoopOutRequest(ctx context.Context, lnd lndclient.LightningClient,
 			errIncorrectChain, chainParams.Name)
 	}
 
+	// Check that the provided destination address is not of type P2PK.
+	switch sweepAddr.(type) {
+	case *btcutil.AddressPubKey:
```

Comment:
Should we instead have a `switch` with all allowed cases (that don't do anything) and then a `default` case that returns a more generic error?
That way we'd also disallow legacy (1xxx) addresses.

Reply:
What are the allowed cases? Everything besides `p2pk`? So `p2pkh` `p2wpkh` `p2wsh` `p2tr`?

Reply:
It looks like we only allow `p2wsh`, `p2wpkh`, `p2sh` and `p2pkh`: https://github.com/lightninglabs/loop/blob/master/sweep/sweeper.go#L108

We'll probably need to add `p2tr` at some point later. We have to match the switch linked to above, otherwise we get the error as seen in the issue.

Reply:
Pushed the changes